frontend/app/src/main/java/com/chads/vanroomies/ChatFragment.java

```diff
@@ -53,11 +74,98 @@ public void onCreate(Bundle savedInstanceState) {
             mParam2 = getArguments().getString(ARG_PARAM2);
         }
     }
-
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        // Inflate the layout for this fragment
-        return inflater.inflate(R.layout.fragment_chat, container, false);
+        View v = inflater.inflate(R.layout.fragment_chat, container, false);
+
+        httpClient = HTTPSClientFactory.createClient(getActivity().getApplication());
+        gson = new Gson();
+        allChatMesssages = new HashMap<>();
+        // TODO: Get userId from backend
+        thisUserId = "654013e0b19c872e994eac8b";
+        chatListRecycler = v.findViewById(R.id.chatlistrecycle);
+
+        setAllChatMesssages(httpClient, getActivity(), v);
+
+        return v;
+    }
+
+    private void setAllChatMesssages(OkHttpClient client, Activity activity, View v) {
+        Request request = new Request.Builder().url(Constants.baseServerURL + Constants.chatsByUserIdEndpoint + thisUserId).build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(@NonNull Call call, @NonNull IOException e) {
+                Log.d(TAG, e.getMessage());
+            }
+
+            @Override
+            public void onResponse(@NonNull Call call, @NonNull Response response) {
+                activity.runOnUiThread(() -> {
+                    try {
+                        String responseData = response.body().string();
+                        Type listType = new TypeToken<List<ChatConversation>>(){}.getType();
+                        Log.d(TAG, "responseData for Conversations is " + responseData);
+                        List<ChatConversation> allConversations = gson.fromJson(responseData, listType);
+
+                        // Iterate through all user conversations
+                        for (ChatConversation conversation : allConversations) {
+                            UserProfile user;
+                            ArrayList<ChatMessage> eachMessageList;
+
+                            // Get userId of both users in a conversation
+                            List<String> userPair = conversation.getUsers();
+                            if (userPair.get(0).equals(thisUserId)) {
+                                user = new UserProfile(userPair.get(1));
+                            }
+                            else {
+                                user = new UserProfile(userPair.get(0));
+                            }
+
+                            eachMessageList = conversation.getMessages();
+
+                            // Check for null userId
+                            if (user == null) {
```

Comment:
I think this should be 
`if (user.getId() == null) {`

user would never be null anyways

Reply:
If we say user is never null, then I don't think user.getId can ever be null either - what do you think?